netsim/modules/_routing.py

```diff
@@ -261,35 +261,61 @@ def remove_vrf_interfaces(node: Box, proto: str) -> None:
-        node.vrfs[l.vrf][proto].active = False
-      node.vrfs[l.vrf][proto] = node[proto] + node.vrfs[l.vrf][proto]       # Add node IGP parameters to VRF IGP parameters
-      node.vrfs[l.vrf][proto].interfaces.append(data.get_box(l))            # Append a copy of the interface data
-      l.pop(proto,None)                                                     # ... and remove global IGP parameters from interface
-                                                                            # Next we need to find if the VRF instance of IGP matters
-      for neighbor in l.neighbors:                                          # ... iterate over the list of neighbors
-        n_data = topology.nodes[neighbor.node]
-        if proto in n_data.get('module',[]):                                # ... and check if at least one of them uses the IGP
-          node.vrfs[l.vrf][proto].active = True
-                                                                            # Cleanup IGP data
-  for vname,vdata in node.get('vrfs',{}).items():                           # ... iterate over the list of VRFs
+    if proto not in l:                                                # Not running the protocol on the interface?
+      continue
+    if 'vrf' not in l:                                                # Interface not in a VRF?
+      continue
+    if node.vrfs[l.vrf][proto] is True:                               # Handle 'force' the protocol by setting it to True
+      node.vrfs[l.vrf][proto] = { 'active': True }
+    elif node.vrfs[l.vrf][proto] is False:                            # Skip protocols disabled on VRF level
+      l.pop(proto,None)
+      continue
+    if not 'active' in node.vrfs[l.vrf][proto]:                       # Assume there are no IGP neighbors in this VRF
+      node.vrfs[l.vrf][proto].active = False
+    node.vrfs[l.vrf][proto] = node[proto] + node.vrfs[l.vrf][proto]   # Add node IGP parameters to VRF IGP parameters
+    data.append_to_list(
+      node.vrfs[l.vrf][proto],
+      list_name='interfaces',
+      item=data.get_box(l))                                           # Append a copy of the interface data
+    if vrf_unnumbered_check and \
+        l.get('ipv4',None) is True and \
+        l.get('_parent_vrf',None) != l.vrf:                           # Have to check the loopback source VRF?
+      data.append_to_list(unnum_err_list,l.vrf,l.ifname)              # ... remember we failed
+
+    l.pop(proto,None)                                                 # Finally, remove global IGP parameters from interface
+                                                                      # Next we need to find if the VRF instance of IGP matters
+    for neighbor in l.neighbors:                                      # ... iterate over the list of neighbors
+      n_data = topology.nodes[neighbor.node]
+      if proto in n_data.get('module',[]):                            # ... and check if at least one of them uses the IGP
```

Comment:
In theory, it could be that 'proto' is disabled on that specific interface on the neighbor side

Reply:
In theory, you might be correct ;) In practice, I don't feel like checking that ;))